### ISSUE TITLE
A test plan cannot be linked to the parent test plan.

### ISSUE THREAD

**Issue by SowmiyaKaruna** [kiwitcms/Kiwi#3802]
### Description of problem
A new test plan cannot be linked to any of the existing test plans. Basically, nesting is not working.

### Version or commit hash (if applicable)
13.5-Enterprise

### Steps to Reproduce

1. Navigate to Testing --> New Test Plan --> Select Product and Enter Test Plan name --> Give the Parent Test Plan ID that we want to link --> Press the save button.
2. Once the test plan has been created, navigate to Edit and look for the parent test plan.
3. Look into the test plan structure.

### Actual results
After Step1: Test plan created with no errors shown.
![TP1](https://github.com/user-attachments/assets/25f48b27-5081-4bc1-9ab8-2416e9a0406c)

After Step2: The parent test plan is empty when we look at the created test plan in edit mode.
![TP2](https://github.com/user-attachments/assets/5723e87a-8202-42d3-aff8-c42f90d945d8)

After Step3: The nested plan structure missing the created test plan.
![TP3](https://github.com/user-attachments/assets/d43b832d-0baa-47bf-a0e0-5a0b68137a16)

Although it is being added to the product, it is not linked to the parent test plan that we have given.
![TP4](https://github.com/user-attachments/assets/af0074c7-0716-4c1b-8aa2-493ec8ba95c9)

### Expected results
Parent test plan should be able to be linked to the new test plans.

**Comment by atodorov** [comment]
Looks like a duplicate of https://github.com/kiwitcms/Kiwi/issues/3765. 


See https://github.com/kiwitcms/Kiwi/issues/3765#issuecomment-2357815449 for more info.
